src/OpenLoco/Windows/ProgressBar.cpp

```diff
@@ -0,0 +1,140 @@
+#include "../Ui/ProgressBar.h"
+#include "../Graphics/Colour.h"
+#include "../Graphics/Gfx.h"
+#include "../Graphics/ImageIds.h"
+#include "../Interop/Interop.hpp"
+#include "../OpenLoco.h"
+#include "../Ui/WindowManager.h"
+#include "../Window.h"
+
+#include <array>
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::Ui::Windows::ProgressBar
+{
+    enum Widx
+    {
+        frame,
+        caption,
+        panel,
+    };
+
+    constexpr Gfx::ui_size_t windowSize = { 350, 47 };
+
+    widget_t widgets[] = {
+        makeWidget({ 0, 0 }, { 350, 47 }, widget_type::frame, 0),
+        makeWidget({ 1, 1 }, { 348, 13 }, widget_type::caption_25, 0, StringIds::buffer_1250),
+        makeWidget({ 0, 15 }, { 350, 32 }, widget_type::panel, 1),
+        widgetEnd(),
+    };
+
+    static window_event_list _events;
+
+    static const char* _captionString;
+    static uint8_t _progressBarStyle = 0; // 0x005233C8
+    static uint8_t _progressBarValue = 0; // 0x011370A8
+
+    void setProgress(uint8_t value);
+    static void initEvents();
+
+    // 0x004CF6E2
+    window* open(const char* captionString)
```

Comment:
could make this a string_view and then make the internal captionString a std::string that way there is no confusion about the owners of the memory as this memory is the buffer from ui/ProgressBar.

Reply:
How would that work with `formatString`? Should I just allocate an empty std::string of size 256 and pass along its buffer?

Reply:
You don't pass _captionString to formatString that was done in the caller of this. A string_view can be constructed from a const char* no bother.